### PR TITLE
Qmaps 685 kebab

### PR DESCRIPTION
**Pull request by Julien-laville**
Add burger menu with configurable item list

### CODE REVIEW

File: config/constants.yml [hunk 1]
```diff
@@ -108,3 +108,51 @@ map:
...
+    - name: _('Press')
+      uri : https://about.qwant.com/press/
+    - name: _('Jobs')
+      uri : https://about.qwant.com/jobs/
+    - name: _('F.A.Q.')
+      uri : https://help.qwant.com/
+    - name: _('Legal notices')
+      uri : https://about.qwant.com/legal/legal-notices/
+    - name: _('Terms of Service')
+      uri : https://about.qwant.com/legal/terms-of-service/qwant-search/ #TODO update to maps
```
Comment (amatissart) [inline config/constants.yml]: will not that URL depend on the user language ? (eg https://about.qwant.com/fr/legal/cgu/qwant-maps/)

File: src/views/menu.dot [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  <div class="menu__panel__top">
+    <i class="icon-x menu__panel__top__close" {{= click(this.close, this) }}></i>
+  </div>
+    <h2 class="menu__panel__section_title">
+      <i class="menu__panel__section__icon icon-map"></i>
+      Qwant Maps
+    </h2>
+    <div class="menu__panel__section">
+      <button class="menu__panel__action" {{= click(this.toggleDirection, this) }}>Itinerary</button>
+      <button class="menu__panel__action" {{= click(this.toggleFavorite, this) }}>Favorites</button>
```
Comment (amatissart) [inline src/views/menu.dot]: Translation keys are missing in this menu

File: src/views/menu.dot [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  <div class="menu__panel">
+  <div class="menu__panel__top">
+    <i class="icon-x menu__panel__top__close" {{= click(this.close, this) }}></i>
+  </div>
+    <h2 class="menu__panel__section_title">
+      <i class="menu__panel__section__icon icon-map"></i>
+      Qwant Maps
+    </h2>
+    <div class="menu__panel__section">
+      <button class="menu__panel__action" {{= click(this.toggleDirection, this) }}>Itinerary</button>
```
Comment (amatissart) [inline src/views/menu.dot]: This button stays active and close the menu on click if "directions" flag is disabled.  
We could disable that button in that case ? Or hide it completely ?

Comment (Julien-laville) [reply src/views/menu.dot]: yep

File: src/views/menu.dot (hunk 2 above)
Comment (Julien-laville) [reply src/views/menu.dot]: fixed

File: config/constants.yml (hunk 1 above)
Comment (Julien-laville) [reply config/constants.yml]: i'll keep url untranslated for now & just add maps term of service

File: src/scss/includes/menu.scss [hunk 4]
```diff
@@ -0,0 +1,116 @@
+.menu__overlay {
+  display: none;
+  height: 100vh;
```
Comment (amatissart) [inline src/scss/includes/menu.scss]: There is a regression on Chrome Android, that might be due to these 100vh values, here and below.  
After opening and closing the burger menu, the page can be scrolled, and a blank zone is visible at the bottom of the screen:

File: src/views/menu.dot [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        <button class="menu__panel__action" {{= click(this.toggleFavorite, this) }}>{{= _('Favorites', 'menu') }}</button>
+      {{?}}
+    </div>
+
+    {{~ this.menuItems:menuItem }}
+      <h2 class="menu__panel__section_title">
+        <i class="menu__panel__section__icon icon-{{= menuItem.icon }}"></i>
+        {{= _(menuItem.sectionName) }}
+      </h2>
+      <div class="menu__panel__section">
```
Comment (amatissart) [inline src/views/menu.dot]: What about the "Contact us" menu ?  
The section name should be a link itself.

File: src/views/menu.dot [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        {{~ menuItem.links:link }}
+          <a class="menu__panel__link" rel="" href="{{= link.uri }}" rel="noopener" target="_blank">
```
Comment (amatissart) [inline src/views/menu.dot]: Remove redundant `rel=""` that takes precedence over "noopener"

File: src/views/menu.dot (hunk 5 above)
Comment (Julien-laville) [reply src/views/menu.dot]: forgot to manage that case

File: src/views/menu.dot (hunk 6 above)
Comment (Julien-laville) [reply src/views/menu.dot]: oups

File: src/views/menu.dot [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+<div class="menu__button" {{= click(this.open, this) }}>
+  <div class="menu__button__line"></div>
+  <div class="menu__button__line"></div>
+  <div class="menu__button__line"></div>
+</div>
+
+<div class="menu">
+  <div class="menu__panel">
+  <div class="menu__panel__top">
+    <i class="icon-x menu__panel__top__close" {{= click(this.close, this) }}></i>
```
[the remark this answers is not in the document]
Comment (Julien-laville) [reply src/views/menu.dot]: fixed it

File: src/views/menu.dot [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+  </div>
+    <h2 class="menu__panel__section_title">
+      <i class="menu__panel__section__icon icon-map"></i>
+      Qwant Maps
+    </h2>
+    <div class="menu__panel__section">
+      <button class="menu__panel__action" {{= click(this.toggleDirection, this) }}>{{= _('Itinerary' , 'menu') }}</button>
+      {{? this.isDirectionActive }}
+        <button class="menu__panel__action" {{= click(this.toggleFavorite, this) }}>{{= _('Favorites', 'menu') }}</button>
+      {{?}}
```
[the remark this answers is not in the document]
Comment (Julien-laville) [reply src/views/menu.dot]: hox is it possible :(

**Pull request merged**